WolvenKit.App/Factories/IChunkViewmodelFactory.cs

```diff
@@ -0,0 +1,13 @@
+﻿using WolvenKit.App.Models.Nodify;
+using WolvenKit.App.ViewModels.Shell;
+using WolvenKit.RED4.Types;
+using WolvenKit.App.ViewModels.Documents;
+
+namespace WolvenKit.App.Factories;
+
+public interface IChunkViewmodelFactory
```

Comment:
I would make a generic class named `IFactory<T> where T : class` and then implement it in the concrete class above.

This `IFactory<T>` can use `ActivatorUtilities.CreateInstance` in a `Create(params object[] args)` method to pass the service provider and extra params.

But as a questions, are these params necessary for the object to be constructed or can these be added later?